test-suite/src/test/groovy/io/micronaut/tracing/jaeger/JaegerConfigurationSpec.groovy

```diff
@@ -22,4 +29,176 @@ class JaegerConfigurationSpec extends Specification {
+     * @author Burt Beckwith
+     * @autohr Manuel Martin
+     */
+    void "test codec configuration W3C B3"() {
+        given:
+        def ctx = ApplicationContext.run(
+                'tracing.jaeger.enabled':'true',
+                'tracing.jaeger.codec.codecs':'W3C,B3' // the order is important
+        )
+        def config = ctx.getBean(JaegerConfiguration.JaegerCodecConfiguration).getCodecConfiguration()
+        Map<Format<?>, List<Codec<TextMap>>> codecs = config.getCodecs()
+        List<Codec<TextMap>> headerCodecs = codecs.get(Format.Builtin.HTTP_HEADERS)
+        List<Codec<TextMap>> textMapCodecs = codecs.get(Format.Builtin.TEXT_MAP)
+
+        println("CODECS=" + codecs)
+        println("HEADER CODECS=" + headerCodecs)
+        println("TEXTMAP CODECS=" + textMapCodecs)
+
+        expect:
+        codecs.size() == 2 // HTTP_HEADERS & TEXT_MAP
+        headerCodecs.size() == 2 // 'TraceContextCodec, B3TextMapCodec'
+        textMapCodecs.size() == 2 // 'TraceContextCodec, B3TextMapCodec'
+        headerCodecs.get(0) instanceof TraceContextCodec // W3C
+        headerCodecs.get(1) instanceof B3TextMapCodec // B3
+        textMapCodecs.get(0) instanceof TraceContextCodec // W3C
+        textMapCodecs.get(1) instanceof B3TextMapCodec // B3
+
+        /*
+        // package io.jaegertracing;
+        public CodecConfiguration withPropagation(Propagation propagation) {
+            switch (propagation) {
+                case JAEGER:
+                    addCodec(codecs, Format.Builtin.HTTP_HEADERS, new TextMapCodec(true));
+                    addCodec(codecs, Format.Builtin.TEXT_MAP, new TextMapCodec(false));
+                    addBinaryCodec(binaryCodecs, Format.Builtin.BINARY, new BinaryCodec());
+                    break;
+                case B3:
+                    addCodec(codecs, Format.Builtin.HTTP_HEADERS, new B3TextMapCodec.Builder().build());
+                    addCodec(codecs, Format.Builtin.TEXT_MAP, new B3TextMapCodec.Builder().build());
+                    break;
+                case W3C: // CHECK THE ONLY WAY YOU CAN GET W3C
+                    addCodec(codecs, Format.Builtin.HTTP_HEADERS, new TraceContextCodec.Builder().build());
+                    addCodec(codecs, Format.Builtin.TEXT_MAP, new TraceContextCodec.Builder().build());
+                    break;
+                default:
+                    log.error("Unhandled propagation format '" + propagation + "'");
+            }
+            return this;
+        }
+        */
+
+        cleanup:
+        ctx.close()
+    }
+
+    /**
+     * @author Burt Beckwith
+     * @autohr Manuel Martin
+     */
+    void "test codec configuration W3C B3 JAEGER"() {
+        given:
+        def ctx = ApplicationContext.run(
+                'tracing.jaeger.enabled':'true',
+                'tracing.jaeger.codec.codecs':'W3C,B3,JAEGER'
+        )
+        def config = ctx.getBean(JaegerConfiguration.JaegerCodecConfiguration).getCodecConfiguration()
```

Comment:
could the subject of test be the `io.jaegertracing.Configuration` aka `JaegerConfiguration#getConfiguration` ?